packages/broker/src/plugins/legacyWebsocket/historicalData.ts

```diff
@@ -62,25 +65,39 @@ export const createResponse = async (
     request: ResendFromRequest|ResendLastRequest|ResendRangeRequest,
     storageNodeRegistry: StorageNodeRegistry
 ): Promise<HistoricalDataResponse> => {
-    const storageNodeUrl = await storageNodeRegistry.getUrlByStreamId(request.streamId)
-    const url = getDataQueryEndpointUrl(request, `${storageNodeUrl}/api/v1`)
-    const abortController = new AbortController()
-    const headers = formAuthorizationHeader(request.sessionToken)
-    const response = await fetch(url, {
-        headers,
-        signal: abortController.signal
-    })
-    if (response.status === 200) {
-        const data = response.body.pipe(split2((message: string) => Protocol.StreamMessage.deserialize(message)))
-        return {
-            data,
-            abort: () => {
-                data.destroy()
-                abortController.abort()
-            },
-            startTime: Date.now()
+    const storageNodeUrls = await storageNodeRegistry.getUrlsByStreamId(request.streamId)
+
+    // Form data query endpoints and shuffle the resulting array
+    const urls = storageNodeUrls.map((storageNodeUrl) => {
```

Comment:
Could use a helper function to shuffle the array, e.g. https://lodash.com/docs/4.17.15#shuffle

Lodash is currently not in `broker`'s dependencies, but IMO we should add it. It's quite "standard" low level JS-library. Once we have that in the toolbox, we are able to focus more to the higher abstraction level 😄

Reply:
I see your point, but I think here it still rather reasonable to determine what is going on. (If you're familiar with comparator stuff)

Reply:
True, I just like to avoid mixing two different abstraction levels within one method. Usually the code is easier to read when there is no mixing. Eg. in this method it is important to know that the array is shuffled, but not important to know how we shuffle it.

In the code there is a comment describing "what is going on". To me it is always a trigger to refactor the logic to helper function 👍